sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -481,7 +482,16 @@ boolean add(Reference symbol) {
 
             private class SymbolToNotNullRangeQueryArgs extends SymbolVisitor<SymbolToNotNullContext, Void> {
 
-                private final Set<String> NULL_CAN_MATCH_FUNCTIONS = ImmutableSet.of("any", "coalesce");
+                private final Set<String> NULL_CAN_MATCH_FUNCTIONS =
```

Comment:
I think the name `NULL_CAN_MATCH_FUNCTIONS` isn't correct in how those functions are used.

In the coalesce case it's correct as it can cause null values to cause a match -> `coalesce(null, 'fallback') = 'fallback' `

But  `select 'foo' = any(x) from t1` will never match if x = null